src/ExpressionEvaluator/Core/Source/ResultProvider/Expansion/TypeVariablesExpansion.cs

```diff
@@ -46,25 +46,24 @@ internal TypeVariablesExpansion(Type declaredType)
             int offset = startIndex2 - index;
             for (int i = 0; i < count2; i++)
             {
-                rows.Add(GetRow(resultProvider, value, i + offset, parent));
+                rows.Add(GetRow(resultProvider, inspectionContext, value, i + offset, parent));
             }
 
             index += _typeArguments.Length;
         }
 
-        private DkmEvaluationResult GetRow(ResultProvider resultProvider, DkmClrValue value, int index, EvalResultDataItem parent)
+        private EvalResultDataItem GetRow(ResultProvider resultProvider, DkmInspectionContext inspectionContext, DkmClrValue value, int index, EvalResultDataItem parent)
         {
-            var inspectionContext = value.InspectionContext;
-            var appDomain = value.Type.AppDomain;
             var typeParameter = _typeParameters[index];
             var typeArgument = _typeArguments[index];
-            var type = DkmClrType.Create(appDomain, typeArgument);
-            var name = typeParameter.Name;
-            var dataItem = new EvalResultDataItem(
-                name,
+            return new EvalResultDataItem(
+                ExpansionKind.TypeVariables,
+                typeParameter.Name,
                 typeDeclaringMember: null,
                 declaredType: typeArgument,
-                value: null,
+                parent: parent,
+                value: value,
+                displayValue: inspectionContext.GetTypeName(DkmClrType.Create(value.Type.AppDomain, typeArgument), Formatter.NoFormatSpecifiers),
```

Comment:
Do these format specifiers need to stay in sync with the ones below?  Extract local?